src/reducers/entities/helpers.js

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+function isPostListNull(pl) {
+    if (!pl) {
+        return true;
+    }
+
+    if (!pl.posts) {
+        return true;
+    }
+
+    return !pl.order;
+}
+
+function makePostListNonNull(pl) {
+    let postList = pl;
+    if (!postList) {
+        postList = {order: [], posts: {}};
+    }
+
+    if (!postList.order) {
+        postList.order = [];
+    }
+
+    if (!postList.posts) {
+        postList.posts = {};
+    }
+
+    return postList;
+}
+
+export function addPosts(state, action) {
+    const newState = {...state};
```

Comment:
This is only shallowly copying the state so whenever we change a child like when we change `newState.latestPageTime` below, we're mutating the state. Other than this function, I think the rest of this file is safe.

It might make sense to write some separate unit tests for this function using deep freeze since it's relatively complicated.